devito/dse/rewriters.py

```diff
@@ -406,6 +408,40 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
         return cluster.rebuild(processed)
 
+    @dse_pass
+    def _skewing(self, cluster, template, **kwargs):
+        """
+        Skew the accesses along the time Dimension.
+        """
+
+        MAX_SKEW_FACTOR = 8
+        """
+        The maximum skew factor applied for the skewing pass.
+        """
+
+        skew_factor = 2  # TOFIX
+
+        if skew_factor not in range(0, MAX_SKEW_FACTOR):
+            raise ValueError("skew_factor must be less than %d" % MAX_SKEW_FACTOR)
+
+        skew_dim, mapper, intervals = None, {}, []
+
+        for i in cluster.ispace.intervals:
+            if i.dim.is_Time:
+                intervals.append(Interval(i.dim, 0, 0))
+                skew_dim = i.dim
+            else:
+                mapper[i.dim] = i.dim + skew_factor*skew_dim
+                intervals.append(Interval(i.dim, -skew_factor*skew_dim,
+                                          -skew_factor*skew_dim))
+
+        # TODO Can I avoid 2 rebuilds?
+        processed = xreplace_indices(cluster.exprs, mapper)
+        cluster.rebuild(processed)
+
+        processed2 = xreplace_indices(cluster.exprs, IntervalGroup(intervals))
```

Comment:
The second argument to ``xreplace_indices`` should **not** be an IntervalGroup.
Even though an exception isn't raised (that's an issue in ``xreplace_indices``), this is conceptually wrong

Reply:
Yes.